client/src/actions/userActions.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * @description User actions for login , signup (register) and logout
+ */
+
+import axios from "axios";
+import {
+  USER_LOGIN_REQUEST,
+  USER_LOGIN_SUCCESS,
+  USER_LOGIN_FAIL,
+  USER_LOGOUT,
+  USER_REGISTER_REQUEST,
+  USER_REGISTER_SUCCESS,
+  USER_REGISTER_FAIL,
+} from "../constants/userConstants";
+
+// User login action
+export const login = async (dispatch, loginPayload) => {
+  const { email, password } = loginPayload;
+
+  try {
+    dispatch({
+      type: USER_LOGIN_REQUEST,
+    });
+
+    const config = {
+      headers: {
+        "Content-Type": "application/json",
+      },
+    };
+
+    const { data } = await axios.post(
+      "http://localhost:3001/api/users/login",
+      { email, password },
+      config
+    );
+
+    dispatch({
+      type: USER_LOGIN_SUCCESS,
+      payload: data,
+    });
+
+    // save data to local storage
+    localStorage.setItem("userInfo", JSON.stringify(data));
+
+    return JSON.stringify(data);
+  } catch (err) {
+    dispatch({
+      type: USER_LOGIN_FAIL,
+      payload:
+        err.response && err.response.data.error
+          ? err.response.data.error
+          : err.message,
+    });
+  }
+};
+
+// Register User Action
+export const register = async (dispatch, registerPayload) => {
+  const { name, email, password } = registerPayload;
+  try {
+    dispatch({
+      type: USER_REGISTER_REQUEST,
+    });
+
+    const config = {
+      headers: {
+        "Content-Type": "application/json",
+      },
+    };
+
+    // data from backend server
+    const { data } = await axios.post(
+      "http://localhost:3001/api/users/register",

```

Comment:
This will break if you deploy, it's much better to use "/api/users/register" and rely on the proxy from the package.json